### PR TITLE
Adds checkstyles to project

### PR DESCRIPTION
**Pull request by onobc**
This commit configures in the Puppycrawl checkstyle with the Spring JavaFormat.

To avoid the intrusive nature of such a change, the spring-grpc-core module is the only one to have the checkstyles applied. I will handle the other modules as time permits.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,6 +10,13 @@ Before submitting a PR, please run the following commands to ensure proper forma
 
 The `-Pjavadoc` is a profile that enables Javadoc processing so as to avoid a long build time when developing.
 
+<dl><dt><strong>📌 NOTE</strong></dt><dd>
```
Comment (onobc) [inline CONTRIBUTING.md:13]: Is this auto-generated from contribution-guidelines.adoc?

File: idea/codeStyleConfig.xml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	</GroovyCodeStyleSettings>
+	<JavaCodeStyleSettings>
+		<option name="CLASS_NAMES_IN_JAVADOC" value="3"/>
+		<option name="INSERT_INNER_CLASS_IMPORTS" value="true"/>
+		<option name="CLASS_COUNT_TO_USE_IMPORT_ON_DEMAND" value="50"/>
+		<option name="NAMES_COUNT_TO_USE_IMPORT_ON_DEMAND" value="500"/>
+		<option name="PACKAGES_TO_USE_IMPORT_ON_DEMAND">
+			<value/>
+		</option>
+		<option name="IMPORT_LAYOUT_TABLE">
```
Comment (onobc) [inline idea/codeStyleConfig.xml:54]: These import order are in line w/ the ones in the checkstyle which allows us to easily optmize imports from w/in IDEA and have them match the checkstyles. 

I am sure there is something similar for other editors. I created a wiki page that tells how to do this for IDEA. W/ a TODO section for other editors.

File: pom.xml [hunk 3]
```diff
@@ -438,6 +488,20 @@
 				<type>pom</type>
 				<scope>import</scope>
 			</dependency>
+			<dependency>
```
Comment (onobc) [inline pom.xml:491]: These were required as the checkstyle enforce AssertJ over Junit assertEquals

File: src/checkstyle/checkstyle.xml [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+			<property name="message"
+					  value="Please use AssertJ imports."/>
+			<property name="ignoreComments" value="true"/>
+		</module>
+		<module name="com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck">
+			<property name="id" value="toLowerCaseWithoutLocale"/>
+			<property name="format" value="\.toLowerCase\(\)"/>
+			<property name="maximum" value="0"/>
+			<property name="message"
+					  value="String.toLowerCase() should be String.toLowerCase(Locale.ROOT)"/>
```
Comment (onobc) [inline src/checkstyle/checkstyle.xml:178]: Here is a bonus for the prevention of the `toLower|upperCase` w/ no locale CVE.

**Comment by dsyer** [comment]
Sigh. I hate these rules. I guess it’s inevitable they will get in here eventually though.

**Comment by onobc** [comment]
> Sigh. I hate these rules. I guess it’s inevitable they will get in here eventually though.

I too have mixed feelings. I end up "accepting" consistency though. 🙁 

Thanks for the review and merge.

**Pull request merged**